src/charm.py

```diff
@@ -57,6 +58,24 @@ def __init__(self, *args):
         self.framework.observe(
             self.alertmanager_lib.cluster_changed, self._on_alertmanager_cluster_changed
         )
+        self.service_hostname = self._external_hostname
+        self.ingress = IngressRequires(
+            self,
+            {
+                "service-hostname": self.service_hostname,
+                "service-name": self.app.name,
+                "service-port": str(self.model.config["port"]),
+            },
+        )
+
+    @property
+    def _external_hostname(self):
+        """Return the external hostname to be passed to ingress via the relation."""
+        # It is recommended to default to `self.app.name` so that the external
+        # hostname will correspond to the deployed application name in the
+        # model, but allow it to be set to something specific via config.
+
+        return self.config["web-external-url"] or f"{self.app.name}.juju"
```

Comment:
Yeah but shouldn't this be consumed in the [`_prometheus_config`](https://github.com/canonical/prometheus-operator/blob/c76f8699f792abd7d8d01023b40ae73f31acf9b8/src/charm.py#L308) method?

Reply:
It should set the Prometheus command line argument for `--web.external-url=<...>` and trigger a restart of Prometheus. The command line arguments are set in `_cli_args()` in `charm.py`.

Reply:
I got it the `web-external-url` work at https://github.com/mmanciop/prometheus-operator/tree/ingress